test/Constraints/patterns.swift

```diff
@@ -21,7 +21,7 @@ default:
 }
 
 switch (1, 2) {
-case (var a, a): // expected-error {{cannot find 'a' in scope}}
+case (var a, a): // expected-error {{cannot find 'a' in scope}} expected-note {{add 'let' or 'var' keyword to bind 'a' in a pattern}}
```

Comment:
Unless I'm mistaken, applying the suggestion here just leads to code that's broken in other ways.

In this case, as in the case of `case none`, it would be better if a non-working or likely unintended suggestion isn't made. It's fine if this PR doesn't include the fix-it to change the latter case to `case .none`, but I think it _is_ important that this PR _not_ suggest something in these cases.

Reply:
Thanks. This is a situation I did not foresee. I'm not keen on fix-its that produce more errors. I'll make a note of this in the original bug and I'll abandon this PR until a proper implementation can be crafted.